libs/components/modals/testing/src/modal-fixture.ts

```diff
@@ -4,53 +4,69 @@ import { ComponentFixture } from '@angular/core/testing';
  * Allows interaction with a SKY UX modal component.
  */
 export class SkyModalFixture {
-  private modalElement: HTMLElement;
+  #modalElement: HTMLElement;
 
-  constructor(private fixture: ComponentFixture<any>, skyTestId: string) {
-    this.modalElement = document.querySelector(
+  #fixture: ComponentFixture<any>;
+
+  constructor(fixture: ComponentFixture<any>, skyTestId: string) {
+    this.#fixture = fixture;
+    const modalElement = document.querySelector(
       'sky-modal[data-sky-id="' + skyTestId + '"]'
-    );
+    ) as HTMLElement;
 
-    if (!this.modalElement) {
+    if (!modalElement) {
       throw new Error(
         `No element was found with a \`data-sky-id\` value of "${skyTestId}".`
       );
+    } else {
+      this.#modalElement = modalElement;
     }
   }
 
   /**
    * The modal component's ARIA describedby attribute.
    */
-  public get ariaDescribedBy(): string {
-    const modalDialogElement = this.getModalDiaglogElement();
-    if (modalDialogElement) {
-      return modalDialogElement.getAttribute('aria-describedby') || undefined;
+  public get ariaDescribedBy(): string | undefined {
+    const modalDialogElement = this.#getModalDiaglogElement();
+    const describedByAttribute =
+      modalDialogElement.getAttribute('aria-describedby');
+    /* We need to have the else case here; however, it wouldn't ever be hit due to our defaults */
+    /* istanbul ignore else */
+    if (describedByAttribute) {
```

Comment:
I get the desire to move the error-throwing to one function, but it appears to confuse TypeScript and/or ESLint as to what the valid code paths are if you have to write code that is unreachable to satisfy it. I'd prefer continuing to throw the error directly in the property and using a constant for the repeated error message than to have to jump through hoops to fool the TS compiler.

Reply:
I don't think it is the error handling as much as it is the fact that we have defaults for the `aria-describedby` if it is undefined by the consumer. I think this is a case where I should have just used the non-null assertion. I'll update that and then we can chat more if needed.

Reply:
If you throw the error directly in the `if` statement, then the code coverage tool will see the `else` case as unreachable and therefore not require you cover it. By throwing an error in an external function, the code coverage tool can't determine that. That's why I'm advocating keeping the `throw` statement local to the function instead of calling a separate function to throw the error.